src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java

```diff
@@ -0,0 +1,231 @@
+package org.prebid.server.bidder.openx;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.Json;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.BidderUtil;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.bidder.openx.model.BidRequestWithError;
+import org.prebid.server.bidder.openx.model.OpenxImpType;
+import org.prebid.server.bidder.openx.proto.ExtImpOpenx;
+import org.prebid.server.bidder.openx.proto.OpenxImpExt;
+import org.prebid.server.bidder.openx.proto.OpenxRequestExt;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+
+/**
+ * OpenX {@link Bidder} implementation.
+ */
+public class OpenxBidder implements Bidder<BidRequest> {
+
+    private static final Logger logger = LoggerFactory.getLogger(OpenxBidder.class);
+
+    private static final String OPENX_CONFIG = "hb_pbs_1.0.0";
+    private static final Function<Imp, OpenxImpType> TYPE_RESOLVER = imp -> imp.getBanner() != null
+            ? OpenxImpType.banner : imp.getVideo() != null ? OpenxImpType.video : OpenxImpType.other;
```

Comment:
I think it is better to replace function static field with static function, and call it like OpenxBidder::resolveType.
Also not sure if we need OpenxImpType enum class just to do this mapping. Can we use strings instead?

Reply:
Agree, having method will save a little memory. 
Regarding enum, I think it's better describes fixed set nature of  possible groupings for impressions in Openx Bidder context.